### DISCUSSION TITLE
Questions from Emacs evil org-mode user

### DISCUSSION THREAD

**Discussion by rieje** [nvim-orgmode/orgmode#241] [Q&A]
As an org-mode user, I'm thinking of migrating to Neovim. I assume contributors to this plugin actually used org-mode, so I was wondering if I can get a similar experience using this plugin (or another Neovim plugin, maybe [Neorg](https://github.com/nvim-neorg/neorg) which seems like a calculated approach for implementing org-mode within the "constraints" of Neovim. Two of the features I use most on org-mode (not necesarily org-mode, but org-mode related):

* `counsel-org-goto-all`: this will popup a small buffer split horizontally where you can fuzzy match headlines of any depth in all your org files and jump to it. A sub-headline is prefixed with the parent headline to provide the user with more context. `swiper` works the same but search candidates are for each line. If I hold a modifier and move up/down to select the line, the buffer will jump to each candidate so I get a true preview before I decide to commit the jump, closing this selection buffer. With this, it's as if all my notes are in one big org file. Of course there's a companion function which does the same but restricts the candidates to be those in the current file.

* [org-super-agenda](https://github.com/alphapapa/org-super-agenda) provides a customizable org-agenda views where you can group items by criteria, e.g. have a group at the top of the agenda showing items due within 7 days, another that groups items by a certain tag, another that sorts the rest of the items by priority. [org-ql](https://github.com/alphapapa/org-ql) is a compatible tool with org-super-agenda that provides advanced searching and filtering. I would at least like to have features of org-super-agenda for custom agendas that allow for grouping sorted items by todo states, tags, priorities, etc. which allow for a GTD workflow. i find the basic org-agenda to be quite limited, restricted only to time-based views when some things are more long-term and don't necessarily have a concrete date.

* `org-emphasis-alist` allows for styling text. For example, __hello__ might show the text as underlined, /hello/ to italicize, **hello** to show the text in a different color, and +hello+ might add a strikethrough, etc. Of course you can toggle this off and see the actual raw text. Copying the text to the system clipboard can exclude the stylizing portion of the text which is the behavior you usually want.
  
* Small miscellaneous features like cycling expanding headlines by levels for a very quick and easy way to visualize structure of notes and possibility of using similar keybindings like C-c C-d, C-c C-w, C-c c, etc.

Does this plugin cover these features? Are there any features that may be difficult to implement in Neovim? Much appreciated.

**Comment by jgollenz** [comment]
Short answer: not yet or only in a limited way.

- counsel-org-goto-all (is this in orgmode core?): no, not at all. I assume if we were to add that, we should utilize the telescope neovim-plugin for that. Sounds doable to me.
- org-super-agenda: no, if it is not in orgmode core, it is generally not planned for this port. However, work on exposing an API for people to recreate such a plugin has recently started.
- org-emphasis-alist: yes, partly. Underline, italicize, bold, code, verbatim and strikethrough are available, but not color. Toggling it off works fine, but copying does not strip the markup. 
- Cycling expanding headlines by levels: I need more info here. Are you talking sub-headings of a single heading or globally in a buffer? I don't know what `C-c C-d`, `C-c C-w`, `C-c c` do, as I'm not an emacs user.

**Comment by pkazmier** [comment]
Like the @rieje , I heavily rely on org-super-agenda as the basic agenda view does allow you to group by criteria. Is there anything else in the ecosystem that would help achieve the same? Would an LLM be able to reproduce org-super-agenda? Anyone try it yet? I’ve got tokens to burn. 🔥 

edit: it already exists apparently! https://github.com/hamidi-dev/org-super-agenda.nvim
